l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/FreezeChain.t.sol

```diff
@@ -0,0 +1,45 @@
+// // SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+import {StateTransitionManagerTest} from "./_StateTransitionManager_Shared.t.sol";
+import {Diamond} from "solpp/state-transition/libraries/Diamond.sol";
+import {DiamondProxy} from "solpp/state-transition/chain-deps/DiamondProxy.sol";
+import {GettersFacet} from "solpp/state-transition/chain-deps/facets/Getters.sol";
+
+contract freezeChainTest is StateTransitionManagerTest {
+    function testFreezingChain() public {
+        address baseToken = address(0x3030303);
+        address sharedBridge = address(0x4040404);
+        address admin = bridgehub;
+        uint256 chainId = block.chainid;
+
+        vm.stopPrank();
+        vm.startPrank(bridgehub);
+
+        chainContractAddress.createNewChain(
+            chainId,
+            baseToken,
+            sharedBridge,
+            admin,
+            abi.encode(getDiamondCutData(diamondInit))
+        );
+
+        address newChainAddress = chainContractAddress.stateTransition(chainId);
+        GettersFacet gettersFacet = GettersFacet(newChainAddress);
+        bool isChainFrozen = gettersFacet.isDiamondStorageFrozen();
+        assertEq(isChainFrozen, false);
+
+        vm.stopPrank();
+        vm.startPrank(governor);
+
+        chainContractAddress.freezeChain(block.chainid);
+
+        // Repeated call should revert
+        vm.expectRevert(bytes.concat("q1")); // storage frozen
+        chainContractAddress.freezeChain(block.chainid);
+
+        // Call fails as storage is frozen
+        vm.expectRevert(bytes.concat("q1"));
+        isChainFrozen = gettersFacet.isDiamondStorageFrozen();
```

Comment:
Is it the case that this getter can't return true? (as the call reverts with the error described above)